akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/InDepthPersistentBehaviorTest.java

```diff
@@ -132,69 +150,105 @@ public PostContent(String postId, String title, String body) {
+            return Effect().persist(event).andThen(() -> cmd.replyTo.tell(Done.getInstance()));
+          })
+          .matchCommand(GetPost.class, (state, cmd) -> {
+            cmd.replyTo.tell(state.postContent);
+            return Effect().none();
+          });
+    }
+
+    private CommandHandlerBuilder<BlogCommand, BlogEvent, BlogState, BlogState> commonCommandHandler() {
+      return commandHandlerBuilder(BlogState.class)
+          .matchCommand(AddPost.class, (state, cmd) -> Effect().unhandled())
+          .matchCommand(PassivatePost.class, (state, cmd) -> Effect().stop());
     }
+    //#post-added-command-handler
 
-    @Override
-    public BlogState emptyState() {
-      return new BlogState(Optional.empty(), false);
-    }
 
-    //#by-state-command-handler
+    //#command-handler
     @Override
     public CommandHandler<BlogCommand, BlogEvent, BlogState> commandHandler() {
-      return byStateCommandHandlerBuilder()
-        .matchState(BlogState.class, (state) -> !state.postContent.isPresent(), initialCommandHandler)
-        .matchState(BlogState.class, (state) -> state.postContent.isPresent(), postCommandHandler)
-        .build();
+      return
+          initialCommandHandler()
+              .orElse(draftCommandHandler())
+              .orElse(publishedCommandHandler())
+              .orElse(commonCommandHandler())
+              .build();
     }
-    //#by-state-command-handler
+    //#command-handler
 
     //#event-handler
     @Override
     public EventHandler<BlogEvent, BlogState> eventHandler() {
       return eventHandlerBuilder()
-        .matchEvent(PostAdded.class, (state, event) -> state.withContent(event.content))
-        .matchEvent(BodyChanged.class, (state, newBody) ->
-          new BlogState(state.postContent.map(pc -> new PostContent(pc.postId, pc.title, newBody.newBody)), state.published))
-        .matchEvent(Published.class, (state, event) -> new BlogState(state.postContent, true))
-        .build();
+          .matchEvent(PostAdded.class, (state, event) ->
+              new DraftState(event.content, false))
+          .matchEvent(BodyChanged.class, DraftState.class, (state, chg) ->
+              state.withContent(new PostContent(state.postId(), state.postContent.title, chg.newBody)))
+          .matchEvent(BodyChanged.class, PublishedState.class, (state, chg) ->
+              state.withContent(new PostContent(state.postId(), state.postContent.title, chg.newBody)))
+          .matchEvent(Published.class, DraftState.class, (state, event) ->
+              new PublishedState(state.postContent))
+          .build();
     }
     //#event-handler
+
+    //#behavior
+    public static Behavior<BlogCommand> behavior(String entityId) {
+      return Behaviors.setup(ctx ->
+          new BlogBehavior("Blog-" + entityId, ctx)
+      );
+    }
+
+    @Override
+    public BlogState emptyState() {
+      return new BlankState();
```

Comment:
EmptyState?

Reply:
was thinking of a blank sheet of paper, which isn't too far from the blogging domain ?